regression/validate-trace-xml-schema/check.py

```diff
@@ -0,0 +1,172 @@
+    # again, intermingles several XML documents
+    'graphml_witness1/test.desc',
+    'switch8/program-only.desc',
+    # show-goto-functions
+    'unsigned1/test.desc',
+    # show-goto-functions
+    'reachability-slice-interproc3/test.desc',
+    'sync_lock_release-1/symbol_per_type.desc'
+]))
+
+if len(sys.argv) != 2:
+    sys.stderr.write('Usage: check.py <path-to-cbmc>')
+CbmcPath = os.path.abspath(sys.argv[1])
+XsdValidateJar = os.path.join(this_script_dir, 'validate-xsd/build/libs/validate-xsd-1.0-SNAPSHOT-uber.jar')
+
+
+class ChangeDir:
+    def __init__(self, change_to_wd):
+        self.current_wd = os.getcwd()
+        self.change_to_wd = change_to_wd
+
+    def __enter__(self):
+        os.chdir(self.change_to_wd)
+
+    def __exit__(self, _type, _value, _tb):
+        os.chdir(self.current_wd)
+
+
+class TestSpec:
+    def __init__(self, args, is_knownbug, is_future, is_thorough):
+        self.args = args
+        self.is_knownbug = is_knownbug
+        self.is_future = is_future
+        self.is_thorough = is_thorough
+
+
+class Validator:
+    def __init__(self, total_specs):
+        self.failed_tests = []
+        self.total_specs = total_specs
+        self.total_test_count = 0
+
+    def check_spec(self, spec_path):
+        self.total_test_count += 1
+        sys.stdout.write('*** Checking [{}/{}] {}... '.format(self.total_test_count, self.total_specs, spec_path))
+        sys.stdout.flush()
+        spec_dir = os.path.dirname(spec_path)
+        spec = self.read_spec(spec_path)
+        if spec.is_knownbug:
+            print('[Skipping KNOWNBUG]')
+            return
+        elif spec.is_future:
+            print('[Skipping FUTURE]')
+            return
+        elif spec.is_thorough:
+            print('[Skipping THOROUGH]')
+            return
+        with ChangeDir(spec_dir):
+            with tempfile.TemporaryFile() as trace_file:
+                self.read_trace_into(trace_file, spec.args)
+                trace_file.seek(0)
+                self.check_trace(spec_path, trace_file)
+
+    def read_trace_into(self, trace_file, args):
+        subprocess.run([CbmcPath, '--trace', '--xml-ui'] + args,
+                       stdout=trace_file)
+
+    def check_trace(self, spec_path, trace_file):
+        validate_result = subprocess.run(['java', '-jar', XsdValidateJar, TraceXsdSpec],
+                                         stdin=trace_file,
+                                         stdout=subprocess.PIPE,
+                                         stderr=subprocess.PIPE)
+        if validate_result.returncode == 0:
+            print('[SUCCESS]')
+        else:
+            print('[FAILURE]')
+            self.failed_tests.append(spec_path)
+        sys.stdout.buffer.write(validate_result.stdout)
+        sys.stdout.buffer.write(validate_result.stderr)
+
+    def read_spec(self, spec_path):
+        with open(spec_path) as spec_file:
+            spec = spec_file.readline().split()
```

Comment:
:bulb: I guess I would use the word `test_kind` rather than spec, which has 3 means in this file:
1. the XSD specificaiton
2. the entire .desc file
3. The first line of the desc file

Reply:
Would like to see this addressed, as I keep thinking `is_knownbug = 'KNOWNBUG' in spec` is a bug, since KNOWNBUG might appear elsewhere in the spec, except here `spec` is just the first line.